cluster/juju/util.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+# Copyright 2014 Canonical LTD. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# If you find any bugs within this script - please file bugs against the
+# Kubernetes Juju Charms project - located here: https://github.com/whitmo/bundle-kubernetes
+
+
+source $KUBE_ROOT/cluster/juju/prereqs/ubuntu-juju.sh
+kube_bundle_url='https://raw.githubusercontent.com/whitmo/bundle-kubernetes/master/bundles.yaml'
+function verify-prereqs() {
+    gather_installation_reqs
+}
+
+function get-password() {
+    echo "TODO: Assign username/password security"
+}
+
+function kube-up() {
+    # If something were to happen that I'm not accounting for, do not
+    # punish the user by making them tear things down. In a perfect world
+    # quickstart should handle this situation, so be nice in the meantime
+    local JUJUSTATUS=$(juju status kubernetes-master --format=oneline)
+    if [[ -z "$JUJUSTATUS" ]]; then
+        if [[ -d "~/.juju/current-env" ]]; then
+            juju quickstart -i -e $kube_jujuenv --no-browser -i $kube_bundle_url
+            # sleeping because of juju bug #
+            sleep 120
+        else
+            juju quickstart --no-browser $kube_bundle_url
+            # sleeping because of juju bug #
+            sleep 120
+        fi
+    fi
+    sleep-status
+}
+
+
+function detect-master() {
+    foo=$(juju status --format=oneline kubernetes-master | cut -d' ' -f3)
+    export KUBE_MASTER_IP=`echo -n $foo`
```

Comment:
Why not just this directly? It doesn't look like that `echo` is really doing anything there...

```shell
  export KUBE_MASTER_IP=${foo}
```

Or just bypass the `${foo}` variable and set `KUBE_MASTER_IP` directly. In that case, keep the `export` statement on a separate line to avoid the problem with interaction of `$(...)` and local/export/declare/etc.